crates/engine/tree/src/tree/precompile_cache.rs

```diff
@@ -18,29 +22,30 @@ impl PrecompileCacheMap {
 /// Cache for precompiles, for each input stores the result.
 #[derive(Debug, Clone)]
 pub struct PrecompileCache(
-    Arc<mini_moka::sync::Cache<CacheKey, CacheEntry, alloy_primitives::map::DefaultHashBuilder>>,
+    Arc<RwLock<HashMap<CacheKey, CacheEntry, alloy_primitives::map::DefaultHashBuilder>>>,
```

Comment:
@Rjected @fgimenez  do we even need moka cache here?

while this impl is still problematic because it is unbounded, I assume a basic rwlock map would also do the job here, and would allow us to not alloc for lookups which is a bit horrible

Reply:
We could try a RwLock LRU map

Reply:
I believe it's unlikely that we will see locks happening at the same time so something like a simple rwlock over an lru that allows us to get rid of the alloc is probably better

@Ayushdubey86 can you change this to parking_lock::RwLock and swap the moka::cache with schnellru::Lrmup, limited Bylength? so `Arc<parking_lot::RwLock<schnellru::LruMap<CacheKey,..>`

Reply:
pushed !:)